packages/gatsby/src/schema/print.ts

```diff
@@ -1,17 +1,44 @@
-const fs = require(`fs-extra`)
-const {
+/* eslint-disable @typescript-eslint/no-use-before-define */
+
+import * as fs from "fs-extra"
+import {
   EnumTypeComposer,
   InputTypeComposer,
   InterfaceTypeComposer,
   ObjectTypeComposer,
   ScalarTypeComposer,
   UnionTypeComposer,
-} = require(`graphql-compose`)
-const report = require(`gatsby-cli/lib/reporter`)
+  SchemaComposer,
+  NamedTypeComposer,
+  ObjectTypeComposerFieldConfigMap,
+  ObjectTypeComposerFieldConfig,
+  ObjectTypeComposerArgumentConfigMap,
+  ObjectTypeComposerArgumentConfig,
+  Extensions,
+  isNamedTypeComposer,
+  EnumTypeComposerValueConfig,
+} from "graphql-compose"
+import report from "gatsby-cli/lib/reporter"
+
+import { internalExtensionNames } from "./extensions"
+import { GraphQLDirective } from "graphql"
 
-const { internalExtensionNames } = require(`./extensions`)
+const {
+  astFromValue,
+  print,
+  GraphQLString,
+  DEFAULT_DEPRECATION_REASON,
+} = require(`graphql`)
+const { printBlockString } = require(`graphql/language/blockString`)
+const _ = require(`lodash`)
 
-const printTypeDefinitions = ({ config, schemaComposer }) => {
+const printTypeDefinitions = ({
+  config,
+  schemaComposer,
+}: {
+  config: any
```

Comment:
Can't we type this?

Reply:
Adjusted in [`77faf7d` (#34787)](https://github.com/gatsbyjs/gatsby/pull/34787/commits/77faf7da06601e1fb6e3204e398980d74f79d3fb)